### PR TITLE
remove old api key format from example docs

### PR DESCRIPTION
**Pull request by Rperry2174**
removes old api key format from examples

### CODE REVIEW

File: examples/rust/rideshare/server/src/main.rs [hunk 1]
```diff
@@ -30,12 +30,16 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+    let auth_user = std::env::var("PYROSCOPE_BASIC_AUTH_USER")
+        .unwrap_or_else(|_| "".to_string());
+
+    let auth_password = std::env::var("PYROSCOPE_BASIC_AUTH_PASSWORD")
         .unwrap_or_else(|_| "".to_string());
 
     // Configure Pyroscope client.
     let agent = PyroscopeAgent::builder(server_address, app_name.to_owned())
-        .auth_token(auth_token)
+        .auth_user(auth_user)
```
Comment (Rperry2174) [inline examples/rust/rideshare/server/src/main.rs:41]: @korniltsev couldn't find docs on this... is this right? we should also add docs

Comment (korniltsev) [reply examples/rust/rideshare/server/src/main.rs:41]: it should be `        .basic_auth("username", "password")`

**Review by korniltsev — APPROVED** [review]
LGTM

note: `PYROSCOPE_AUTH_TOKEN` is used by the og pyroscope examples cluster ingesting data to pyroscope cloud. once we merge the PR it will be harder to update og pyroscope examples cluster

**Review by knylander-grafana — APPROVED** [review]
Thank you for updating the java.md file.

**Pull request merged**